### PR TITLE
Add CD-grid stress arrays

### PR DESCRIPTION
**Pull request by phil-blain**
to be discussed... I tried grepping 'stressp_1' and added the new arrays where I thought they were needed... I'll clean up the commit history tomorrow. Probably missing some stuff

### CODE REVIEW

**Review by eclare108213 — COMMENTED** [review]
Looks good so far.  Thanks for implementing these arrays in the restart files!

File: cicecore/cicedynB/analysis/ice_history.F90 [hunk 1]
```diff
@@ -4287,17 +4288,29 @@ subroutine accum_hist (dt)
...
+            case('CD')
+               call principal_stress (nx_block,  ny_block,  &
+                                      stresspT  (:,:,iblk), &
+                                      stressmT  (:,:,iblk), &
+                                      stress12T (:,:,iblk), &
+                                      strength  (:,:,iblk), &
+                                      sig1      (:,:,iblk), &
+                                      sig2      (:,:,iblk), &
+                                      sigP      (:,:,iblk))
+            case('CD')
```
Comment (dabail10) [inline cicecore/cicedynB/analysis/ice_history.F90]: Why is there an extra case('CD') here?

File: cicecore/cicedynB/general/ice_flux.F90 [hunk 2]
```diff
@@ -135,6 +135,9 @@ module ice_flux
          stressp_1, stressp_2, stressp_3, stressp_4 , & ! sigma11+sigma22
          stressm_1, stressm_2, stressm_3, stressm_4 , & ! sigma11-sigma22
          stress12_1,stress12_2,stress12_3,stress12_4    ! sigma12
```
Comment (dabail10) [inline cicecore/cicedynB/general/ice_flux.F90]: This line needs a comma and ampersand.

**Review by dabail10 — CHANGES_REQUESTED** [review]
This is looking really good. There might be some syntax errors beyond the one that github actions failed on.

File: cicecore/cicedynB/analysis/ice_history.F90 (hunk 1 above)
Comment (phil-blain) [reply cicecore/cicedynB/analysis/ice_history.F90]: fixed.

File: cicecore/cicedynB/general/ice_flux.F90 (hunk 2 above)
Comment (phil-blain) [reply cicecore/cicedynB/general/ice_flux.F90]: also fixed

**Comment by phil-blain** [comment]
Yeah I pushed that without even compiling (!). I compiles now, I'll update the PR soon.

**Comment by phil-blain** [comment]
I cleaned up the compile failures, and cleaned up the commit history.

There are ~two~ three TODOs that maybe we will want to discuss..

File: cicecore/cicedynB/infrastructure/ice_restart_driver.F90 [hunk 3]
```diff
@@ -465,6 +495,7 @@ subroutine restartfile (ice_ic)
             stress12_4(i,j,iblk) = c0
          enddo
          enddo
+         ! TODO: CD-grid ?
```
Comment (phil-blain) [inline cicecore/cicedynB/infrastructure/ice_restart_driver.F90:498]: These initialization date from https://github.com/CICE-Consortium/CICE-svn-trunk/commit/788626f9

**Review by dabail10 — APPROVED** [review]

**Review by apcraig — APPROVED** [review]

**Review by eclare108213 — APPROVED** [review]

**Pull request merged**